homeassistant/components/snips.py

```diff
@@ -59,21 +61,55 @@ def message_received(topic, payload, qos):
             _LOGGER.error('Intent has invalid schema: %s. %s', err, request)
             return
 
-        intent_type = request['intent']['intentName'].split('__')[-1]
+        snips_response = None
+
+        if request['intent']['intentName'].startswith('user'):
+            intent_type = request['intent']['intentName'].split('__')[-1]
+        else:
+            intent_type = request['intent']['intentName'].split(':')[-1]
         slots = {}
         for slot in request.get('slots', []):
-            if 'value' in slot['value']:
-                slots[slot['slotName']] = {'value': slot['value']['value']}
-            else:
-                slots[slot['slotName']] = {'value': slot['rawValue']}
+            slots[slot['slotName']] = {'value': resolve_slot_values(slot)}
 
         try:
-            yield from intent.async_handle(
+            intent_response = yield from intent.async_handle(
                 hass, DOMAIN, intent_type, slots, request['input'])
+            if 'plain' in intent_response.speech:
+                snips_response = intent_response.speech['plain']['speech']
+        except intent.UnknownIntent as err:
+            _LOGGER.warning("Received unknown intent %s",
+                            request['intent']['intentName'])
+            snips_response = "Unknown Intent"
         except intent.IntentError:
             _LOGGER.exception("Error while handling intent: %s.", intent_type)
+            snips_response = "Error while handling intent"
+
+        notification = {'sessionId': request.get('sessionId', 'default'),
+                        'text': snips_response}
+
+        _LOGGER.debug("send_response %s", json.dumps(notification))
+        mqtt.async_publish(hass, 'hermes/dialogueManager/endSession',
+                           json.dumps(notification))
 
     yield from hass.components.mqtt.async_subscribe(
         INTENT_TOPIC, message_received)
 
     return True
+
+
+def resolve_slot_values(slot):
+    """Convert snips builtin types to useable values."""
+    if 'value' in slot['value']:
+        value = slot['value']['value']
+    else:
+        value = slot['rawValue']
+
+    if slot.get('entity') == "snips/duration":
+        delta = timedelta(weeks=slot['value']['weeks'],
+                          days=slot['value']['days'],
+                          hours=slot['value']['hours'],
+                          minutes=slot['value']['minutes'],
+                          seconds=slot['value']['seconds'])
+        value = delta.seconds
```

Comment:
I think that you mean to call `delta.total_seconds()`, otherwise you might as well just return `slot['value']['seconds']`

Reply:
total_seconds() is fine but returns a float value which is not needed since the precision isn't there. The values are discrete so slot values minutes=5, seconds=0 would returns 0 seconds.

Reply:
Oh, but I see what you mean in that seconds only includes up to 24 hours worth of seconds, fixed.